### ISSUE TITLE
Ads are not showing on the real device

### ISSUE THREAD

**Issue by tminet** [AppLovin/AppLovin-MAX-SDK-Android#103]
I updated a old project from Java to Kotlin and Jetpack Compose.
While running the app in an emulator they work, but switching to a real device, doesn't work.

I'm currently using the newest version of the AppLovin SDK (11.0.0), I also tried with the previous SDK version (10.3.5) and the results were the same. The type of ads I'm trying to display are Interstitials (I haven't done the test with others yet, like Rewarded, but I will do as soon as possible).

Testing so far has been done using both standard builds, debug (with minify and shrink disabled) and release (with minify and shrink enabled). And as mentioned before, Test Ads are displayed normally when the device is an emulator, but on a real device (currently tried on 4 different devices) they are not displayed.

Following logcat, the only information that seems to be relevant are these:
```
W/System.err:     at com.applovin.impl.sdk.utils.Utils.getAlwaysFinishActivitiesSetting()
W/System.err:     at com.applovin.impl.mediation.ads.MaxFullscreenAdImpl.a()
W/System.err:     at com.applovin.impl.mediation.ads.MaxFullscreenAdImpl.showAd()
W/System.err:     at com.applovin.mediation.ads.MaxInterstitialAd.showAd()
W/System.err:     at com.applovin.mediation.ads.MaxInterstitialAd.showAd()
```
They happen at the exact moment the ad should be displayed.

Does anyone have any information on what the problem could be?

**Comment by santoshbagadi** [comment]
Hi @tminet , do you by any chance have the `Don't Keep Activities` developer setting enabled on your device? That setting is currently not supported by our SDK. If it is enabled, could you please disable it and see if you are able to show the ad.

**Comment by tminet** [comment]
Hello @santoshbagadi.
`Don't Keep Activities` is not enabled here. In the old version of the app, with Java, the ads are displayed normally. And also the other apps I have with Kotlin and XML, it works.

Adding more information than I've tested so far:

Rewarded Ads are not showing even on the virtual device. Several times the following error appears in logcat
`[AppLovinFullscreenActivity] Encountered media error: Video view error (-38,0) for ad: AppLovinAd{adIdNumber=6146428, source=DECODED_AD_TOKEN_JSON, zoneId="null"}`
I also noticed that this same error **appears a few times** with Interstitial Ads too.

Which made me think that some Interstitial Ads and all Rewarded Ads might require some dependency to be displayed, that I'm not using, like the `androidx.appcompat:appcompat:$latestVer` for `AppCompatActivity()`. But adding it again had no effect.
After this test it made me think that maybe the problem could be in the fact that my activity uses the `ComponentActivity()` (since it is a project with Compose) and Activity's is required for `MaxRewardedAd.getInstance("AD_UNIT", activity)` and `MaxInterstitialAd("AD_UNIT", activity)` so... I have no idea why, I'm just thinking of theories after the tests I did.

I don't know if the current AppLovin SDK is compatible with Jetpack Compose, from what I looked for, I didn't find anyone saying anything about using the two together...

**Comment by santoshbagadi** [comment]
@tminet , could you please share the full logs with our SDKs verbose logging enabled, so that we can investigate further.

**Comment by tminet** [comment]
@santoshbagadi I think now it's functional here, it's not 100%, but it's working...

I believe the problem was in the way I implemented the `MaxAdListener`, before i was using separate class as use case and was injecting the Activity as dependency with Hilt. Now with the implementation directly in the Activity worked.

I uploaded the source code [here](https://github.com/tminet/AppLovin-Compose) of how I did the implementation at the moment.
